docs/roadmap.md

```diff
@@ -0,0 +1,83 @@
+Kargo's roadmap
+=================
+
+### Self deployment (pull-mode)
+- the playbook would install and configure docker and the etcd cluster
+- the following data would be inserted into etcd: certs,tokens,users,inventory,group_vars.
+- a "kubespray" container would be deployed (kargo-cli, ansible-playbook, kpm)
+- to be discussed, a way to provide the inventory
+- self deployment of the node
+
+### Provisionning and cloud providers
+- Terraform to provision instances on GCE, AWS, Openstack, Digital Ocean, Azure
+- On AWS autoscaling, multi AZ, multi-region (Ubernetes)
+- On Azure autoscaling, create loadbalancer 
+- On GCE be able to create a loadbalancer automatically (IAM ?)
+- TLS boostrap support for kubelet 
+  (related issues: https://github.com/kubernetes/kubernetes/pull/20439 <br>
+   https://github.com/kubernetes/kubernetes/issues/18112)
+
+### Tests
+- Run kubernetes e2e tests
+- CI Tests for CoreOS (CentOS Support = requirements for us)
+- migrate to jenkins
+(a test is currently a deployment on a 3 node cluste, testing k8s api, ping between 2 pods)
+- Full tests on GCE per day (All OS's, all network plugins)
+- trigger a single test per pull request
```

Comment:
@mattymo We should consider using a _Jenkinsfile_ (https://jenkins.io/doc/pipeline/jenkinsfile/) which is somehow equivalent to the _.travis.yml_ file. Moreover Jenkins is able to detect new branches / PR in the repo and spawn new jobs for building those branches https://jenkins.io/blog/2015/12/03/pipeline-as-code-with-multibranch-workflows-in-jenkins/).